zio-schema/shared/src/main/scala/zio/schema/validation/PhoneNumberValidation.scala

```diff
@@ -0,0 +1,2383 @@
+    val phoneNumber = digitsWithSeparator.between(9, 10)
+
+    Validation.regex(prefix ~ optionalSeparator ~ phoneNumber)
+  }
+
+  /** Phone number validation for Gabon */
+  lazy val phoneNumberGA: Validation[String] = {
+    val countryCode = Regex.literal("241")
+    val prefix      = internationalPrefix ~ optionalSeparator ~ countryCode | nationalPrefixZero
+    val phoneNumber = digitsWithSeparator.between(9, 10)
+
+    Validation.regex(prefix ~ optionalSeparator ~ phoneNumber)
+  }
+
+  /** Phone number validation for United Kingdom */
+  lazy val phoneNumberGB: Validation[String] = {
+    val countryCode = Regex.literal("44")
+    val prefix      = internationalPrefix ~ optionalSeparator ~ countryCode | nationalPrefixZero
+    val phoneNumber = digitsWithSeparator.between(9, 10)
+
+    Validation.regex(prefix ~ optionalSeparator ~ phoneNumber)
+  }
+
+  /** Phone number validation for Grenada */
+  lazy val phoneNumberGD: Validation[String] = {
+    val countryCode         = Regex.literal("1")
+    val internationalPrefix = plus | Regex.literal("011")
+    val prefix              = internationalPrefix ~ optionalSeparator ~ countryCode | nationalPrefixOne
+    val leadingDigits       = Regex.literal("473")
+    val phoneNumber         = digitsWithSeparator.between(6, 7)
+
+    Validation.regex(prefix ~ optionalSeparator ~ leadingDigits ~ optionalSeparator ~ phoneNumber)
+  }
+
+  /** Phone number validation for Georgia */
+  lazy val phoneNumberGE: Validation[String] = {
+    val countryCode = Regex.literal("995")
+    val prefix      = internationalPrefix ~ optionalSeparator ~ countryCode | nationalPrefixZero
+    val phoneNumber = digitsWithSeparator.between(9, 10)
+
+    Validation.regex(prefix ~ optionalSeparator ~ phoneNumber)
+  }
+
+  /** Phone number validation for French Guiana */
+  lazy val phoneNumberGF: Validation[String] = {
+    val countryCode = Regex.literal("594")
+    val prefix      = internationalPrefix ~ optionalSeparator ~ countryCode | nationalPrefixZero
+    val phoneNumber = digitsWithSeparator.between(9, 10)
+
+    Validation.regex(prefix ~ optionalSeparator ~ phoneNumber)
+  }
+
+  /** Phone number validation for Guernsey */
+  lazy val phoneNumberGG: Validation[String] = {
+    val countryCode = Regex.literal("44")
+    val prefix      = internationalPrefix ~ optionalSeparator ~ countryCode | nationalPrefixZero
+    val phoneNumber = digitsWithSeparator.between(9, 10)
+
+    Validation.regex(prefix ~ optionalSeparator ~ phoneNumber)
+  }
+
+  /** Phone number validation for Ghana */
+  lazy val phoneNumberGH: Validation[String] = {
+    val countryCode = Regex.literal("233")
+    val prefix      = internationalPrefix ~ optionalSeparator ~ countryCode | nationalPrefixZero
+    val phoneNumber = digitsWithSeparator.between(9, 10)
+
+    Validation.regex(prefix ~ optionalSeparator ~ phoneNumber)
+  }
+
+  /** Phone number validation for Gibraltar */
+  lazy val phoneNumberGI: Validation[String] = {
```

Comment:
I see that this is a very common pattern, probably would worth moving into a method with parameters for the country code and maybe the min/max number of digits to reduce code duplication.

(or maybe a couple of such methods if there are other common cases shared by many countries)

The more unique cases could remain as they are, of course.

Reply:
Good question, the point here is that each specific country has its own dialing rules and its own prefixes, in fact, the original issue only asked for US, UK and Europe. For example, as it stands, we can add whatever rules we want to the US and UK individually, but I don't really know what they want the scope of this validation to be.  I'm going to refactor the code, but adding new specific validations for each country would lead to the same problem.